### ISSUE TITLE
[Bug] Integration is not always resiliant to printer power being cycled and needs to be reloaded to work again

### ISSUE THREAD

**Issue by rjbutler** [greghesp/ha-bambulab#294]
### Describe the bug

If I turn off the printer (say to swap the nozzle), the integration sometimes seems to stop working after I turn it back on and I can no longer control the lights/fans.  Reloading the integration re-enables functionality.  This isn't consistent -- sometimes it works fine.

### To Reproduce

1. Turn off printer
2. Turn it back on
3. Try to control accessories using the integration
4. 

### Expected Behaviour

The integration should recover after a power cycle of the printer

### What device are you using?

P1P

### Diagnostic Output

```shell
{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2023.11.3",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.11.6",
    "docker": true,
    "arch": "x86_64",
    "timezone": "America/Los_Angeles",
    "os_name": "Linux",
    "os_version": "6.1.59",
    "supervisor": "2023.11.3",
    "host_os": "Home Assistant OS 11.1",
    "docker_version": "24.0.6",
    "chassis": "embedded",
    "run_as_root": true
  },
  "custom_components": {
    "bambu_lab": {
      "version": "1.9.1",
      "requirements": []
    },
    "moonraker": {
      "version": "0.10.0",
      "requirements": [
        "moonraker-api==2.0.5"
      ]
    },
    "hass_agent": {
      "version": "2022.11.9",
      "requirements": []
    },
    "chime_tts": {
      "version": "v0.8.0",
      "requirements": [
        "pydub"
      ]
    },
    "spook": {
      "version": "1.3.2",
      "requirements": []
    },
    "vesync": {
      "version": "1.3.0",
      "requirements": [
        "pyvesync==2.1.10"
      ]
    },
    "presence_simulation": {
      "version": "3.2",
      "requirements": []
    },
    "mass": {
      "version": "2023.2.2",
      "requirements": []
    },
    "rainforest_emu_2": {
      "version": "1.3.2",
      "requirements": [
        "pyserial-asyncio==0.6"
      ]
    },
    "hacs": {
      "version": "1.33.0",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    },
    "tapo": {
      "version": "2.11.0",
      "requirements": [
        "plugp100==3.13.1"
      ]
    },
    "hubitat": {
      "version": "0.9.8",
      "requirements": []
    },
    "zha_toolkit": {
      "version": "v1.1.6",
      "requirements": [
        "pytz"
      ]
    },
    "ha_registry": {
      "version": "1.0.0",
      "requirements": []
    },
    "resmed_myair": {
      "version": "0.1.11",
      "requirements": [
        "beautifulsoup4",
        "PyJWT"
      ]
    },
    "fontawesome": {
      "version": "2.2.1",
      "requirements": []
    },
    "alarmo": {
      "version": "v1.9.13",
      "requirements": []
    },
    "govee_ble_lights": {
      "version": "0.1.1",
      "requirements": [
        "bleak-retry-connector"
      ]
    },
    "spook_inverse": {
      "version": "1.3.2",
      "requirements": []
    }
  },
  "integration_manifest": {
    "domain": "bambu_lab",
    "name": "Bambu Lab",
    "codeowners": [
      "@greghesp",
      "@AdrianGarside"
    ],
    "config_flow": true,
    "dependencies": [
      "device_automation",
      "mqtt",
      "ffmpeg"
    ],
    "documentation": "https://github.com/greghesp/ha-bambulab",
    "iot_class": "local_push",
    "issue_tracker": "https://github.com/greghesp/ha-bambulab/issues",
    "ssdp": [
      {
        "st": "urn:bambulab-com:device:3dprinter:1"
      }
    ],
    "version": "1.9.1",
    "is_built_in": false
  },
  "data": {
    "push_all": {
      "upload": {
        "status": "idle",
        "progress": 0,
        "message": ""
      },
      "nozzle_temper": 23.3125,
      "nozzle_target_temper": 0,
      "bed_temper": 22.3125,
      "bed_target_temper": 0,
      "chamber_temper": 5,
      "mc_print_stage": "1",
      "heatbreak_fan_speed": "0",
      "cooling_fan_speed": "0",
      "big_fan1_speed": "0",
      "big_fan2_speed": "0",
      "mc_percent": 0,
      "mc_remaining_time": 0,
      "ams_status": 0,
      "ams_rfid_status": 6,
      "hw_switch_state": 1,
      "spd_mag": 100,
      "spd_lvl": 2,
      "print_error": 0,
      "lifecycle": "product",
      "wifi_signal": "-32dBm",
      "gcode_state": "IDLE",
      "gcode_file_prepare_percent": "0",
      "queue_number": 0,
      "queue_total": 0,
      "queue_est": 0,
      "queue_sts": 0,
      "project_id": "0",
      "profile_id": "0",
      "task_id": "0",
      "subtask_id": "0",
      "subtask_name": "",
      "gcode_file": "",
      "stg": [],
      "stg_cur": 0,
      "print_type": "idle",
      "home_flag": 17808,
      "mc_print_line_number": "0",
      "mc_print_sub_stage": 0,
      "sdcard": true,
      "force_upgrade": false,
      "mess_production_state": "active",
      "layer_num": 0,
      "total_layer_num": 0,
      "s_obj": [],
      "fan_gear": 0,
      "hms": [],
      "online": {
        "ahb": false,
        "rfid": false,
        "version": 391537706
      },
      "ams": {
        "ams": [
          {
            "id": "0",
            "humidity": "5",
            "temp": "0.0",
            "tray": [
              {
                "id": "0",
                "remain": -1,
                "k": 0.014999999664723873,
                "n": 1,
                "tag_uid": "0000000000000000",
                "tray_id_name": "",
                "tray_info_idx": "GFL99",
                "tray_type": "PLA",
                "tray_sub_brands": "",
                "tray_color": "00000000",
                "tray_weight": "0",
                "tray_diameter": "0.00",
                "tray_temp": "0",
                "tray_time": "0",
                "bed_temp_type": "0",
                "bed_temp": "0",
                "nozzle_temp_max": "240",
                "nozzle_temp_min": "190",
                "xcam_info": "000000000000000000000000",
                "tray_uuid": "00000000000000000000000000000000"
              },
              {
                "id": "1",
                "remain": -1,
                "k": 0.019999999552965164,
                "n": 1,
                "tag_uid": "0000000000000000",
                "tray_id_name": "",
                "tray_info_idx": "GFL99",
                "tray_type": "PLA",
                "tray_sub_brands": "",
                "tray_color": "F4D976FF",
                "tray_weight": "0",
                "tray_diameter": "0.00",
                "tray_temp": "0",
                "tray_time": "0",
                "bed_temp_type": "0",
                "bed_temp": "0",
                "nozzle_temp_max": "240",
                "nozzle_temp_min": "190",
                "xcam_info": "000000000000000000000000",
                "tray_uuid": "00000000000000000000000000000000"
              },
              {
                "id": "2",
                "remain": -1,
                "k": 0.019999999552965164,
                "n": 1,
                "tag_uid": "0000000000000000",
                "tray_id_name": "",
                "tray_info_idx": "GFL99",
                "tray_type": "PLA",
                "tray_sub_brands": "",
                "tray_color": "893044FF",
                "tray_weight": "0",
                "tray_diameter": "0.00",
                "tray_temp": "0",
                "tray_time": "0",
                "bed_temp_type": "0",
                "bed_temp": "0",
                "nozzle_temp_max": "240",
                "nozzle_temp_min": "190",
                "xcam_info": "000000000000000000000000",
                "tray_uuid": "00000000000000000000000000000000"
              },
              {
                "id": "3",
                "remain": -1,
                "k": 0.019999999552965164,
                "n": 1,
                "tag_uid": "0000000000000000",
                "tray_id_name": "",
                "tray_info_idx": "GFG99",
                "tray_type": "PETG",
                "tray_sub_brands": "",
                "tray_color": "FFFFFFFF",
                "tray_weight": "0",
                "tray_diameter": "0.00",
                "tray_temp": "0",
                "tray_time": "0",
                "bed_temp_type": "0",
                "bed_temp": "0",
                "nozzle_temp_max": "270",
                "nozzle_temp_min": "220",
                "xcam_info": "000000000000000000000000",
                "tray_uuid": "00000000000000000000000000000000"
              }
            ]
          }
        ],
        "ams_exist_bits": "1",
        "tray_exist_bits": "f",
        "tray_is_bbl_bits": "f",
        "tray_tar": "255",
        "tray_now": "1",
        "tray_pre": "1",
        "tray_read_done_bits": "f",
        "tray_reading_bits": "0",
        "version": 3,
        "insert_flag": true,
        "power_on_flag": true
      },
      "ipcam": {
        "ipcam_dev": "1",
        "ipcam_record": "enable",
        "timelapse": "disable",
        "mode_bits": 3
      },
      "vt_tray": {
        "id": "254",
        "tag_uid": "0000000000000000",
        "tray_id_name": "",
        "tray_info_idx": "GFL95",
        "tray_type": "PLA",
        "tray_sub_brands": "",
        "tray_color": "BCBCBCFF",
        "tray_weight": "0",
        "tray_diameter": "0.00",
        "tray_temp": "0",
        "tray_time": "0",
        "bed_temp_type": "0",
        "bed_temp": "0",
        "nozzle_temp_max": "240",
        "nozzle_temp_min": "190",
        "xcam_info": "000000000000000000000000",
        "tray_uuid": "00000000000000000000000000000000",
        "remain": 0,
        "k": 0,
        "n": 1
      },
      "lights_report": [
        {
          "node": "chamber_light",
          "mode": "off"
        }
      ],
      "upgrade_state": {
        "sequence_id": 0,
        "progress": "",
        "status": "",
        "consistency_request": false,
        "dis_state": 0,
        "err_code": 0,
        "force_upgrade": false,
        "message": "",
        "module": "",
        "new_version_state": 2,
        "new_ver_list": []
      },
      "command": "push_status",
      "msg": 0,
      "sequence_id": "116"
    },
    "get_version": {
      "command": "get_version",
      "sequence_id": "21450",
      "module": [
        {
          "name": "ota",
          "project_name": "C12",
          "sw_ver": "01.04.02.00",
          "hw_ver": "OTA",
          "sn": "**REDACTED**"
        },
        {
          "name": "esp32",
          "project_name": "C12",
          "sw_ver": "01.05.16.97",
          "hw_ver": "AP04",
          "sn": "**REDACTED**"
        },
        {
          "name": "mc",
          "project_name": "P1",
          "sw_ver": "00.00.16.49",
          "loader_ver": "00.00.00.00",
          "hw_ver": "MC07",
          "sn": "**REDACTED**"
        },
        {
          "name": "th",
          "project_name": "P1",
          "sw_ver": "00.00.05.83",
          "loader_ver": "00.00.00.00",
          "hw_ver": "TH09",
          "sn": "**REDACTED**"
        },
        {
          "name": "ams/0",
          "project_name": "",
          "sw_ver": "00.00.06.40",
          "loader_ver": "00.00.00.00",
          "ota_ver": "00.00.00.00",
          "hw_ver": "AMS08",
          "sn": "**REDACTED**"
        }
      ],
      "result": "success",
      "reason": ""
    }
  }
}
```

### Log Extracts

```shell
I do see these errors in the log, I assume it's related:

This error originated from a custom integration.

Logger: custom_components.bambu_lab.pybambu
Source: custom_components/bambu_lab/pybambu/bambu_client.py:307
Integration: Bambu Lab (documentation, issues)
First occurred: November 25, 2023 at 6:58:06 PM (1 occurrences)
Last logged: November 25, 2023 at 6:58:06 PM

An exception occurred processing a message:
This error originated from a custom integration.

Logger: custom_components.bambu_lab.pybambu
Source: custom_components/bambu_lab/pybambu/bambu_client.py:308
Integration: Bambu Lab (documentation, issues)
First occurred: November 25, 2023 at 6:58:06 PM (1 occurrences)
Last logged: November 25, 2023 at 6:58:06 PM

Exception type: <class 'RuntimeError'>

This error originated from a custom integration.

Logger: custom_components.bambu_lab.pybambu
Source: custom_components/bambu_lab/pybambu/bambu_client.py:309
Integration: Bambu Lab (documentation, issues)
First occurred: November 25, 2023 at 6:58:06 PM (1 occurrences)
Last logged: November 25, 2023 at 6:58:06 PM

Exception args: ('list changed size during iteration',)
```

**Comment by AdrianGarside** [comment]
I've not seen this at all with my P1P. Can you enable debug logging for the integration, get a repro and provide the debug log output that includes the context leading up that error. It's coming from processing the mqtt payload from the printer but there's a lot of code involved in that and I'm hoping more complete debug log data will narrow down where that runtime error is coming from.

**Comment by AdrianGarside** [comment]
Probably dupe of #256

**Comment by AdrianGarside** [comment]
Closing as a dupe of 256. I've made a mitigation fix in:
https://github.com/greghesp/ha-bambulab/releases/tag/v2.0-dev2

That might help here.

**Comment by AdrianGarside** [comment]
Above wouldn't help. This should actually fix it:
https://github.com/greghesp/ha-bambulab/releases/tag/v2.0-dev5

**Comment by AdrianGarside** [comment]
Fixed in v2.0.0.
